### ISSUE TITLE
[java] JUnitUseExpected should not trigger if exception is used

### ISSUE THREAD

**Issue by huxi** [pmd/pmd#4660]
**Affects PMD Version:** 
6.55

**Rule:**

[JUnitUseExpected](https://docs.pmd-code.org/pmd-doc-6.55.0/pmd_rules_java_bestpractices.html#junituseexpected)

**Description:**
I routinely check that the content of caught exceptions contains expected values, i.e. that exceptions are thrown as expected.

This isn't possible using `@Test(expected=SomeRuntimeException.class)`

**Code Sample demonstrating the issue:**

```java
try {
    instance.accept(event);
    fail("Expected exception wasn't thrown!");
} catch (SomeRuntimeException ex) {
    assertEquals(
            "expected message",
            ex.getMessage()
    );
    var cause = ex.getCause();
    assertNotNull(cause);
    assertSame(
            IllegalArgumentException.class,
            cause.getClass()
    );
    assertEquals(
            "argument foo was invalid",
            cause.getMessage()
    );
}
```

**Expected outcome:**

PMD should not report a violation if the caught exception is used to assert things.

**Running PMD through:** *[Maven | Other]*

**Comment by huxi** [comment]
Sorry. My bad. The reported issue was valid and caused by a different catch in the same method.
